src/components/schedules/sidebar/Modules.jsx

```diff
@@ -2,71 +2,170 @@ import React from "react";
 import Nav from "react-bootstrap/Nav";
 import { scroller } from "react-scroll";
 
-// not being used at the moment
-function Modules({ scheduleData }) {
-  const moduleArray = [];
+function Modules({ scheduleData, coursetype }) {
+  const moduleNameArray = [];
+  const scheduleUrls = [];
+  const scheduleObjs = [];
 
-  // getting names of all modules
   Object.keys(scheduleData).map((day) => {
     if (scheduleData[day].dateTypes.module) {
-      if (!moduleArray.includes(scheduleData[day].dateTypes.module)) {
-        moduleArray.push(scheduleData[day].dateTypes.module);
-      }
+      Object.keys(scheduleData[day].dateTypes)
```

Comment:
I'd like to see this refactored so that:

1. it's in a helper function
2. it's less nested - maybe by using `map`s that call helper functions?
3. contains comments